exchanges/okx/okx_test.go

```diff
@@ -3319,23 +3319,21 @@ func TestUpdateTradablePairs(t *testing.T) {
 
 func TestUpdateOrderExecutionLimits(t *testing.T) {
 	t.Parallel()
-
 	testexch.UpdatePairsOnce(t, e)
 	for _, a := range e.GetAssetTypes(false) {
-		err := e.UpdateOrderExecutionLimits(contextGenerate(), a)
-		if !assert.NoError(t, err) {
-			continue
-		}
+		t.Run(a.String(), func(t *testing.T) {
+			t.Parallel()
+			require.NoError(t, e.UpdateOrderExecutionLimits(contextGenerate(), a), "UpdateOrderExecutionLimits must not error")
 
-		p, err := e.GetAvailablePairs(a)
-		require.NoErrorf(t, err, "GetAvailablePairs for asset %s must not error", a)
-		require.NotEmptyf(t, p, "GetAvailablePairs for asset %s must not return empty pairs", a)
+			p, err := e.GetAvailablePairs(a)
+			require.NoErrorf(t, err, "GetAvailablePairs for asset %s must not error", a)
```

Comment:
🚧 🌴  Also using `*f` inside a Run
```suggestion
			require.NoError(t, err, "GetAvailablePairs must not error")
```

Reply:
What makes this an issue when iterating over things? You don't know where the issue occurs
We have 135 instances of NoErrorf for assertify and some of them introduced by you.
What is the rule? Is it for the test setup? ie GetAvailablePairs being the problem because it is not the main purpose of the test? Or is it because you want less detail in a loop?

Reply:
I think we're on the same page. I'll just restate it from my view and if I don't say anything you don't agree with then we're good 👍 
* Users should be able to understand the failure by combining the test name with the message
* Assertions should have a description unless the function called was the same name as the unit test, and it's clear what's being tested and why
* ErrorIs doesn't need a description if it's unique and clear enough on it's own

I think maybe we're being a bit more anal about format for these right now to establish good precedents throughout the codebase so that contributors have a good chance of copy/pasting the right thing.

Noting: I'm only a `t.Run` convert because of you 😛